### ISSUE TITLE
Files on shares bug leading to wider data corruption 

### ISSUE THREAD

**Issue by Stefan-mcp** [giuspen/cherrytree#667]
when working on a file located in a share, the host lost connection to the share. 
Cherrytree started losing the content of nodes in the tree, though the node structure remained seemingly intact. as more node were explorered more content disappeared. Attempting to save to a local drive at this point to try rescue as much data as possible resulted write error. Unforturnately i had chosen to try update an old copy that was on the local disk and though the write generated an error, the file i had chosen was still replaced with a 40kb blank one. attempting to save to any other location gave the same outcome. 
Running out of options i decided to try export what content remained from the current cherrytree instances to anything, just to save it. i was able to export to a plaintext file successfully, but the content was mainly missing with just a few nodes having their content in place, the rest being blank. Additionally, in this export, some of the tree structure had disappeared towards the bottom.

I conclude from this experience, 
1 - working on a share is not a good idea in the current version 
2 - cherrytree doesnt notice is working on a share and doesnt use any local storage backup, it just seems to be holding content in ram.
3 - cherrytree does create temp files (with ~, ~~, ~~~)  endings, and looks to be using these as the workspace for open instances of a file - but it doesnt realise when these arent local
4 - when these ~ working files become inaccessible cherrytree starts behaving strangely, but theres no alert, error raised - the situ is not being caught by any error handler
5 - because the error hasnt been detected due to 4, cherrytrees in memory document starts to get silently mangled, again, no error is thrown at this point, despite the failed reads that i assume must be happening in the background
6 - after the initial write error, the write function seems to get corrupted, affecting future writes that should be possible. Alternatively it could be there is some underlying read thats failinug at this point an the error message is non descript stating a write error, when the underlying cause is actually a read error as cherrytree tries to parse the in memory document with the ~ temps files to produce the output

**Comment by Stefan-mcp** [comment]
tldr: 
tmp files arent stored in /tmp - they are relative to the path of the document and this can have unforseen consequences and generally bork the code. 
error checking routines around ~ reads could probably be improved

**Comment by txe** [comment]
It should be fixed in the last version (0.39.0) by #662

**Comment by Stefan-mcp** [comment]
Thanks, looking at your commit can see you have implementing reconnecting logic which i assume mitigates some of the issue. However, consider the case where the storage doesnt come back. The real issue here seems more that the temp files live with the document - the user is essentially a client to a remote db and doesnt have a complete copy of their data locally on guaranteed storage to allow them to recover should the remote storage go away. Could I recommend instead that we address the issue by storing the temp files, which essentially are the current instance version, locally. That way remote storage,  USB, whatever, will never impact the users ability to save the current instance. 
Further to this, I suspect there may be security implications with storing the temp files remotely, potentially permissions issues, and the possibility to leverage the temp files to perform some sort of injection attack against a cherrytree client that's using them trustingly. 
Imagine someone was to softlink or switch out the db whilst some root user elsewhere is using it, I see mischief potential
